yt_dlp/extractor/instagram.py

```diff
@@ -58,14 +58,18 @@ def _perform_login(self, username, password):
             self._LOGIN_URL, None, note='Downloading login webpage', errnote='Failed to download login webpage')
 
         shared_data = self._parse_json(self._search_regex(
-            r'window\._sharedData\s*=\s*({.+?});', login_webpage, 'shared data', default='{}'), None)
+            r'\["XIGSharedData",\s*\[\],\s*({.+?}),\d+\]', login_webpage, 'shared data', default='{}'), None)
```

Comment:
Support the old pattern too

Reply:
@pukkandan The old pattern doesn't seem to work at all anymore, why support it?